lib/server.js

```diff
@@ -914,73 +900,92 @@ function createServer(options, clients) {
+                                        handlerErr, req.s3Request, req);
+
+                                    /*
+                                     * Get proper status code
+                                     * (ObjectNotFoundError  should be 404)
+                                     */
+                                    var statusCode = 500; // default
+                                    if (handlerErr.name ===
+                                        'ObjectNotFoundError' ||
+                                        handlerErr.restCode ===
+                                        'ObjectNotFoundError') {
+                                            statusCode = 404;
+                                    } else if (handlerErr.name ===
+                                               'BucketNotFoundError' ||
+                                               handlerErr.restCode ===
+                                               'BucketNotFoundError') {
+                                                   statusCode = 404;
+                                    } else if (handlerErr.name ===
+                                               'BucketExistsError' ||
+                                               handlerErr.restCode ===
+                                               'BucketExistsError') {
+                                                   statusCode = 409;
+                                    } else if (handlerErr.statusCode &&
+                                               typeof (handlerErr. statusCode)
+                                               === 'number') {
+                                                   statusCode =
+                                                       handlerErr.statusCode;
+                                    } else if (handlerErr.code &&
+                                               typeof (handlerErr.code) ===
+                                               'number') {
+                                        statusCode = handlerErr.code;
+                                    }
+
+                                    res.writeHead(statusCode, {
+                                        'Content-Type': 'application/xml',
+                                        'Content-Length':
+                                        Buffer.byteLength(s3XmlError, 'utf8'),
+                                        'x-amz-request-id':
+                                        res.getHeader('x-amz-request-id') ||
+                                            'unknown',
+                                        'x-amz-id-2':
+                                            res.getHeader('x-amz-id-2') ||
+                                                'unknown'});
+
+                                    res.write(s3XmlError, 'utf8');
+                                    res.end();
                                     next(false); // Stop route processing
```

Comment:
This `next(false); // Stop route processing` and the one in the else case... can't they just be pulled out  into one line after the if, and the else{} on 990-991 just goes away?